storage/src/main/java/io/atomix/storage/buffer/FileBytes.java

```diff
@@ -297,6 +298,22 @@ public Bytes read(int position, byte[] bytes, int offset, int length) {
     return this;
   }
 
+  @Override
+  public Bytes read(int position, ByteBuffer bytes, int offset, int length) {
+    checkRead(position, length);
+
+    try {
+      final ByteBuffer dest = bytes.slice();
```

Comment:
does it make sense to slice here?

Reply:
I usually slice to avoid mutating the given `ByteBuffer` as that can result in unexpected consequences from the calling code. What do you think?

Reply:
Not 100% familiar with the api but it seems that it will create a new byte buffer. Does it also copy the complete content?

Reply:
A slice is just a view of the buffer, it shares the same contents as the parent buffer, but has a different position/mark/limit/etc.

Reply:
ok thanks